bin/ch_fuse.c

```diff
@@ -0,0 +1,157 @@
+/* Copyright © Triad National Security, LLC, and others. */
+
+#define _GNU_SOURCE
+#include <fcntl.h>
+#include <grp.h>
+#include <libgen.h>
+#include <pwd.h>
+#include <sched.h>
+#include <semaphore.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/mount.h>
+#include <sys/prctl.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <time.h>
+#include <unistd.h>
+
+//new libs
+#include <dirent.h>
+#include <sys/wait.h>
+#include <ll.h>
+
+#include "ch_fuse.h"
+#include "ch_core.h"
+#include "ch_misc.h"
+
+/** Macros **/
+
+/** Types **/
+struct squash {
+   char *mountdir;      // mount location of sqfs
+   pid_t pid;           // ID of fuse loop
+   sqfs_ll_chan chan;   // fuse channel associated with squash fuse session
+   sqfs_ll *ll;         // squashfs image
+};
+
+/** Constants **/
+
+/** Global variables **/
+struct squash sq;
+
+/** Function prototypes (private) **/
+void get_fuse_ops(struct fuse_lowlevel_ops *sqfs_ll_ops);
+void kill_fuse();
+
+/** Functions **/
+
+/* Initalize fuse loop */
+void fuse_loop_init()
+{
+   Ze((fuse_set_signal_handlers(sq.chan.session) == -1), "failed to set signal handlers");
+   if ((fork()) != 0) { //parent process
+      sq.pid = getpid();
+      Ze((fuse_session_loop(sq.chan.session) == -1), "failed to create fuse loop");
+      exit(0);
+   }
+}
+
+/* Function called to start sqfs_ll_clean */
+void kill_fuse()
+{
+   DEBUG("end fuse loop: %d", sq.pid);
+   exit(0);
+}
+
+/* Exit handler for sqfs */
+void sqfs_ll_clean()
```

Comment:
Don't use the `sqfs_ll` prefix because it implies the functions are in the SquashFUSE library (it's an informal namespace).